Libraries/PeriphDrivers/Source/SPIMSS/spimss_reva.c

```diff
@@ -231,6 +241,203 @@ int MXC_SPIMSS_RevA_MasterTrans(mxc_spimss_reva_regs_t *spi, spimss_reva_req_t *
+    MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(rx_channel_id), MXC_DMA_Handler);
+
+    states[spi_num].req = req;
+
+    // TX dma configuration settings.
+    dma_config_tx.ch = tx_channel_id;
+    dma_config_tx.reqsel = MXC_DMA_REQUEST_SPI1TX;
+    dma_config_tx.dstwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.srcwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_tx.dstinc_en = 0; // Ignored since the destination is set to SPIMSS TX fifo.
+    dma_config_tx.srcinc_en = 1;
+
+    // advanced config for tx channel
+    dma_adv_config_tx.ch = tx_channel_id;
+    dma_adv_config_tx.burst_size = req->bits < 16 ? 1 : 2;
+
+    // advanced config for rx channel
+    dma_adv_config_rx.ch = rx_channel_id;
+    dma_adv_config_rx.burst_size = req->bits < 16 ? 1 : 2;
+
+    dma_config_rx.ch = rx_channel_id;
+    dma_config_rx.reqsel = MXC_DMA_REQUEST_SPI1RX;
+    dma_config_rx.dstwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_rx.srcwd = MXC_DMA_WIDTH_HALFWORD;
+    dma_config_rx.dstinc_en = 1;
+    dma_config_rx.srcinc_en = 0; // Ignored since the source is set to SPIMSS RX fifo.
+
+    //Setting TX configuration values for this spi transaction.
+    srcdst_config_tx.ch = tx_channel_id;
+    srcdst_config_tx.len = sent_byte_len;
+    srcdst_config_tx.source = req->tx_data;
+    MXC_DMA_ConfigChannel(dma_config_tx, srcdst_config_tx);
+    MXC_DMA_AdvConfigChannel(dma_adv_config_tx);
+
+    // Enable TX channel CTZ interrupt to be sure that TX operation is completed
+    MXC_DMA_ChannelClearFlags(tx_channel_id,
+                              MXC_DMA_ChannelGetFlags(tx_channel_id)); // Clear interrupts.
+    MXC_DMA_EnableInt(tx_channel_id); // Enable DMA peripheral
+    //interrupt for txChannel.
+
+    MXC_DMA_SetCallback(tx_channel_id, MXC_SPIMSS_RevA_DMA_Handler);
+    MXC_DMA_SetChannelInterruptEn(tx_channel_id, false, true); // Enable ctz interrupt.
+    MXC_DMA_Start(tx_channel_id);
+
+    // rx src dst config.
+    srcdst_config_rx.ch = rx_channel_id;
+    srcdst_config_rx.len = sent_byte_len;
+    srcdst_config_rx.dest = req->rx_data;
+    MXC_DMA_ConfigChannel(dma_config_rx, srcdst_config_rx);
+    MXC_DMA_AdvConfigChannel(dma_adv_config_rx);
+
+    //Enable RX channel CTZ interrupt to be sure that RX operation is completed.
+    MXC_DMA_ChannelClearFlags(rx_channel_id,
+                              MXC_DMA_ChannelGetFlags(rx_channel_id)); // Clear interrupts.
+    MXC_DMA_EnableInt(rx_channel_id); // Enable DMA peripheral
+    //interrupt for txChannel.
+    MXC_DMA_SetCallback(rx_channel_id, MXC_SPIMSS_RevA_DMA_Handler);
+    MXC_DMA_SetChannelInterruptEn(rx_channel_id, false, true); // Enable ctz interrupt.
+    MXC_DMA_Start(rx_channel_id);
+
+    spi->dma |= MXC_F_SPIMSS_REVA_DMA_TX_DMA_EN; // Enable TX DMA requests
+    spi->dma |= MXC_F_SPIMSS_REVA_DMA_RX_DMA_EN; // Enable RX DMA requests
+
+    spi->mode &= ~(MXC_F_SPIMSS_REVA_MODE_SSV); // Set Slave Select to LOW.
+    spi->ctrl |= (MXC_F_SPIMSS_REVA_CTRL_ENABLE); // Start transaction.
```

Comment:
I'm just wondering, but do you know how long the setup time is from the moment the SPIMSS DMA Transaction function is called to the moment the transaction begins?

Reply:
I started a timer in the example code and get the timer counter both at the beginning of MXC_SPIMSS_MasterTransDMA function and at the end of the same function after running the line -> "spi->ctrl |= (MXC_F_SPIMSS_REVA_CTRL_ENABLE); ". According to my measurement, it does not take around even 1 ms on average in debug mode. I am sharing the timer settings and the screenshot of the measurement. Please let me know if something is wrong about the measurement method and results.

Here is the settings for timer:
uint32_t periodTicks = MXC_TMR_GetPeriod(MXC_TMR0, 1, 1);
cfg.pres = TMR_PRES_1;
cfg.mode = TMR_MODE_CONTINUOUS;
cfg.cmp_cnt = periodTicks; 
cfg.pol = 0;

MXC_TMR_Init(MXC_TMR0, &cfg);
MXC_TMR_Start(MXC_TMR0);

This is the SS of the measurement:
![TimerCapture](https://github.com/Analog-Devices-MSDK/msdk/assets/134288264/5284ddb9-d722-4db9-9f6c-ea9e1652d6ec)

Reply:
@sihyung-maxim Looks like around 88uS.  Good enough for a first pass I think.  @kilicomercan FYI it's better than our old SPI v1 drivers (~135uS).  SPI v2 drivers have gotten it down to ~20uS.

We can take a look at applying the SPI v2 optimizations when we tackle porting v2 to the other parts.